tictactoe/src/main/java/models/TicTacToeService.java

```diff
@@ -0,0 +1,144 @@
+package main.java.models;
+
+import java.util.List;
+
+public class TicTacToeService {
+  private final GameBoard gameboard = new GameBoard();
+
+  public void setNoOfColumnsGameBoard(int setNoOfColumnsGameBoard) {
+    gameboard.setNoOfColumnsGameBoard(setNoOfColumnsGameBoard);
+  }
+
+  public void setNoOfRowsGameBoard(int setNoOfRowsGameBoard) {
+    gameboard.setNoOfRowsGameBoard(setNoOfRowsGameBoard);
+  }
+
+  public void setPlayers(List<Player> players) {
+    gameboard.setPlayers(players);
+  }
+
+  public void setUserRowColumnInputs(List<UserRowColumnInput> userRowColumnInputs) {
+    gameboard.setUserRowColumnInputs(userRowColumnInputs);
+  }
+
+  public void setGrid(int noOfRowsGameBoard, int noOfColumnsGameBoard) {
+    gameboard.setGrid(noOfRowsGameBoard,noOfColumnsGameBoard);
+  }
+
+  public void startGame() {
+
+    String [][]grid = gameboard.getGrid();
+    initialiseGameBoard(grid);
+    displayGrid(grid);
+    List<Player> playerQueue;
+    playerQueue = gameboard.getPlayers();
+    for(int k=0;k<gameboard.getUserRowColumnInputs().size();k++) {
+      Player currentPlayer = playerQueue.get(0);
+      playerQueue.remove(0);
+      int inputRowNum = gameboard.getUserRowColumnInputs().get(k).getInputRowNum();
+      int inputColumnNum = gameboard.getUserRowColumnInputs().get(k).getInputColumnNum();
+      if(!isValidMove(grid,inputRowNum,inputColumnNum)) {
+        System.out.println("Invalid Move");
+        playerQueue.add(0,currentPlayer);
+      }
+      else {
+        grid[inputRowNum-1][inputColumnNum-1] = currentPlayer.getChosenCharacter()+" ";
+        displayGrid(grid);
+        if(isGameComplete(grid)) {
+          System.out.println("Game Over");
+          break;
+        }
+        if(rowCheck(grid,inputRowNum,inputColumnNum) || columnCheck(grid,inputRowNum,inputColumnNum)
+            || leftDiagonalCheck(grid,inputRowNum,inputColumnNum) || rightDiagonalCheck(grid,inputRowNum,inputColumnNum)) {
+          System.out.println(currentPlayer.getName()+" won the game");
+          break;
+        }
+        playerQueue.add(currentPlayer);
+      }
+    }
+  }
+
+  private void displayGrid(String[][] grid) {
+    for (String[] strings : grid) {
+      for (String string : strings) {
```

Comment:
string?

Reply:
It was shown in the suggestion of the IDE